tests/layer/test_deep_graph_infomax.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from stellargraph.layer import DeepGraphInfoMax, GCN
+from stellargraph.mapper import FullBatchNodeGenerator, CorruptedGenerator
+from ..test_utils.graphs import example_graph_random
+import tensorflow as tf
+import pytest
+
+
+def test_dgi():
```

Comment:
Given we now have a basic unit test, do you think #1003 is basically just "add tests for DGI with GAT, PPNP, APPNP" now?

Reply:
Yeah I think #1003 is adding tests for GAT/PPNP/APPNP along with maybe testing some edge cases. It shouldn't be too much work and I'm happy to add them in there if that would easier review wise.

Reply:
Nah, unless you're really enthusiastic, this PR has already been open long enough, let's not add another chunk of new code to review.